cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -269,11 +273,20 @@ Secret brokersInternalSecret() {
                             kafka.getLogging() instanceof ExternalLogging ?
                                     configMapOperations.get(assemblyCm.getMetadata().getNamespace(), ((ExternalLogging) kafka.getLogging()).getName()) :
                                     null);
+                    StatefulSet ss = kafka.generateStatefulSet(isOpenShift);
+                    if (oldKafkaCM != null && !oldKafkaCM.getData().get(ANCILLARY_CM_KEY_LOG_CONFIG).toString().equals(logAndMetricsConfigMap.getData().get(ANCILLARY_CM_KEY_LOG_CONFIG))) {
+                        log.debug("Kafka configmap change detected. Restarting pods");
+                        ss.getSpec().getTemplate().getMetadata().getAnnotations().put(ANNOTATION_GENERATION, "-1");
+                    } else {
+                        log.debug("Kafka fresh start");
+                    }
+
                     KafkaClusterDescription desc =
                             new KafkaClusterDescription(kafka, kafka.generateService(), kafka.generateHeadlessService(),
-                                    logAndMetricsConfigMap, kafka.generateStatefulSet(isOpenShift),
+                                    logAndMetricsConfigMap, ss,
```

Comment:
I think what you need to do is use the return value from the reconciliation of the log and metrics config map to inform whether a roll is necessary by the `maybeRollPods()`. I think it would be OK for `maybeRollPods()` to take a `boolean forceRoll` parameter to do that.